### PR TITLE
[Docs] Fix code snippets in "How to write code snippets"

### PR DESCRIPTION
**Pull request by bveeramani**
## Why are these changes needed?

This PR fixes two issues.

---

**Issue 1**: This guide contains `doctest` and `testcode` blocks that are explicitly written out for illustrative purposes.  For example:

image

These blocks aren't tested by our current tooling (`sphinx.ext.doctest`), but they will be tested by our new tooling (`pytest-sphinx`).

---

**Issue 2:** Datasets created from ragged images produce non-deterministic representations. 

This is because the ndarrray shape varies depending on which images are read. For example:
```
>>> ray.data.read_images("s3://anonymous@air-example-data-2/imagenet-sample-images")
Dataset(
   num_blocks=14,
   num_rows=14,
   schema={image: numpy.ndarray(shape=(300, 400, 3), dtype=uint8)}  # (300, 400, 3) is non-deterministic  
)
```

To avoid non-deterministic examples, this PR updates the guide to read fixed-sized images.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by angelinalg — APPROVED** [review]

File: doc/source/ray-contribute/writing-code-snippets.rst [hunk 1]
```diff
@@ -226,21 +228,21 @@ the `doctest` directive and replace problematic sections with ellipsis. ::
         :options: +ELLIPSIS
 
         >>> import ray
-        >>> ray.data.read_images("s3://anonymous@air-example-data-2/imagenet-sample-images")
+        >>> ray.data.read_images("example://image-datasets/simple")
```
Comment (matthewdeng) [inline doc/source/ray-contribute/writing-code-snippets.rst:231]: Can you explain in the PR description why this change was needed?

Comment (bveeramani) [reply doc/source/ray-contribute/writing-code-snippets.rst:231]: Done

**Review by amogkam — APPROVED** [review]
